src/Compilers/CSharp/Test/Semantic/Semantics/IndexAndRangeTests.cs

```diff
@@ -1550,5 +1550,87 @@ static Index Create(out Index y)
                 //         var result = y..Create(out Index y);
                 Diagnostic(ErrorCode.ERR_VariableUsedBeforeDeclaration, "y").WithArguments("y").WithLocation(7, 22));
         }
+
+        [Fact, WorkItem(39852, "https://github.com/dotnet/roslyn/issues/39852")]
+        public void DontAllowNamedArgumentsForImplicitRangeIndexer()
+        {
+            CreateCompilationWithIndexAndRangeAndSpan(@"
+using System;
+public class C 
+{
+    public static void M(string text) 
+    {
+        _ = text[startIndex: 1..^1];
+        _ = new Span<char>(text.ToCharArray())[start: 1..^1];
+        _ = text[range: 1..^1];
+    }
+}").VerifyDiagnostics(
+                    // (7,18): error CS8761: Implicit Range indexer cannot have a named argument.
+                    //         _ = text[startIndex: 1..^1];
+                    Diagnostic(ErrorCode.ERR_ImplicitIndexerWithName, "startIndex").WithArguments("Range").WithLocation(7, 18),
+                    // (8,48): error CS8761: Implicit Range indexer cannot have a named argument.
+                    //         _ = new Span<char>(text.ToCharArray())[start: 1..^1];
+                    Diagnostic(ErrorCode.ERR_ImplicitIndexerWithName, "start").WithArguments("Range").WithLocation(8, 48),
+                    // (9,18): error CS8761: Implicit Range indexer cannot have a named argument.
+                    //         _ = text[range: 1..^1];
+                    Diagnostic(ErrorCode.ERR_ImplicitIndexerWithName, "range").WithArguments("Range").WithLocation(9, 18));
+        }
+
+        [Fact, WorkItem(39852, "https://github.com/dotnet/roslyn/issues/39852")]
+        public void DontAllowNamedArgumentsForImplicitIndexIndexer()
+        {
+            CreateCompilationWithIndexAndRangeAndSpan(@"
+using System;
+public class C 
+{
+    public static void M(string text) 
+    {
+        _ = text[index: ^1];
```

Comment:
>index [](start = 17, length = 5)

Does this name match underlying parameter name? Consider testing with a "wrong" name as well.